cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1996,6 +2072,60 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
 
     // Sync before calling the next `encode_pages` which may alter the stats data.
     if (need_sync) { stream.synchronize(); }
+
+    // now add to the column chunk SizeStatistics if necessary
+    if (stats_granularity == statistics_freq::STATISTICS_COLUMN) {
```

Comment:
Yeah, I suppose...just need to sync earlier then.

Reply:
Just trying to figure out if we can avoid this `save_r` business. Could it at least have a descriptive name?

Reply:
I left the loops as is, because I didn't want to have to sync twice. I instead renamed the outer loop's `r` and then use that for each of the inner loops over `r`.